tests/validation/tests/v3_api/test_secrets.py

```diff
@@ -623,12 +623,33 @@ def create_project_client(request):
     namespace["project"] = p
     namespace["c_client"] = c_client
 
+
     def fin():
         client = get_user_client()
         client.delete(namespace["project"])
+
     request.addfinalizer(fin)
 
 
+def rancher_version_check():
+    # Checks if rancher version is greater than v2.6
+    # eg: v2.5-head, v2.5 etc.
+
+    rancher_version = get_setting_value_by_name('server-version')
+
+    version_check = False
+    if rancher_version.startswith('v'):
```

Comment:
version.parse() should be able to handle version constructs that match [PEP 440](https://www.python.org/dev/peps/pep-0440/#version-scheme). So only handling the "-head" case is necessary:
```
if "head" in rancher_version:
            rancher_version = ''.join(rancher_version.split("-")[0])
OR
# this is more explicit imo
if "head" in rancher_version:
            rancher_version = rancher_version.replace("-head")
```

Reply:
@git-ival replacing `"-head"` will result in an error as the v2.5-head versions have commit ID attached. 
eg:`v2.5-47e2c3e7324fb74c643ec877db0272ab500b5097-head` which gives the output as `v2.5-47e2c3e7324fb74c643ec877db0272ab500b5097`.

Reply:
That makes sense, but do we need the top-level `if rancher_version.startswith('v'):` ?

Reply:
I added the startswith('v') as we have master-head as well as one of the versions. But now that I think about it, since the master-head version will not be covered anymore, we can remove the startwith('v') validation.

Reply:
Hm, I think that `version.parse()` would error on "master-head" since it doesn't match the expected pattern. If we ever do have to test against master-head we should explicitly check for it and return `True` if so.

Reply:
I think its ok to have the extra validation for the rancher version (startswith('v'). If we ever come back to master it will not break and no additional changes will be needed.

For```Hm, I think that version.parse() would error on "master-head" ```
I removed it from the previous feedback from Brandon as currently we won't be using it and might switch back to `main` or `master` in future.